tests/unit/backend/test_common.py

```diff
@@ -45,3 +45,97 @@ class TestBackend(PyRDF.backend.Backend.Backend):
 
         with self.assertRaises(TypeError):
             TestBackend()
+
+class IncludeHeadersTest(unittest.TestCase):
+    """
+    Tests to check the working of 'PyRDF.Include' function.
+
+    """
+    class TestBackend(PyRDF.backend.Backend.Backend):
+        """
+        A Test Backend class to check the working of
+        'PyRDF.Include' function.
+
+        """
+        def execute(self, generator):
+            pass
+
+    def test_default_empty_list_include(self):
+        """
+        Test case to ensure that 'PyRDF.Include' function
+        defaults to an empty list if no parameter is given.
+
+        """
+        PyRDF.current_backend = IncludeHeadersTest.TestBackend()
+
+        PyRDF.include()
+
+        self.assertListEqual(PyRDF.current_backend.includes, [])
+
+    def test_string_include(self):
+        """
+        Test case to check the working of 'PyRDF.Include'
+        function when a single string is passed to it.
+
+        """
+        PyRDF.current_backend = IncludeHeadersTest.TestBackend()
+
+        PyRDF.include("header1")
+
+        self.assertListEqual(PyRDF.current_backend.includes, ["header1"])
+
+    def test_list_include(self):
+        """
+        Test case to check the working of 'PyRDF.Include'
+        function when a list of strings is passed to it.
+
+        """
+        PyRDF.current_backend = IncludeHeadersTest.TestBackend()
+
+        PyRDF.include(["header1"])
+
+        self.assertListEqual(PyRDF.current_backend.includes, ["header1"])
+
+    def test_list_extend_include(self):
+        """
+        Test case to check the working of 'PyRDF.Include'
+        function when different lists of strings are passed
+        to it multiple times.
+
+        """
+        PyRDF.current_backend = IncludeHeadersTest.TestBackend()
+
+        PyRDF.include(["header1", "header2"])
+        PyRDF.include(["header3", "header4", "header5"])
+
+        self.assertListEqual(PyRDF.current_backend.includes, ["header1", "header2", "header3", "header4", "header5"])
+
+class DeclareHeadersTest(unittest.TestCase):
+    """
+    Tests to check the working of the static
+    method 'declare_headers' in Backend class.
+
+    """
+    def test_single_header_declare(self):
+        """
+        Test case to check the working of 'declare_headers'
+        function when a single header needs to be included.
+
+        """
+        PyRDF.backend.Backend.Backend.declare_headers(["tests/unit/backend/test_headers/header1.cpp"])
```

Comment:
Can this be `PyRDF.current_backend.declare_headers`?

Reply:
We will also need to test this functionality with all the backends that we implement (Local, Spark), to make sure that the declared headers are taken when running a computation with that backend.

Reply:
`current_backend` refers to an instance of `Backend` @etejedor . But `declare_headers` is a static method. So that wouldn't work right ?

Reply:
Nop sorry. My bad. It'd still work @etejedor

Reply:
@etejedor , can I write a couple more integration tests for checking each backend after this PR ?

Reply:
What if then you find a problem with your code when writing the integration tests? :) I think you can include it here, just take the time that you need.

Reply:
Ok works @etejedor . In that case, can I reuse an existing integration test or tutorial and only change one function call to a header include ?